torch/csrc/utils/python_numbers.h

```diff
@@ -45,6 +45,14 @@ inline int64_t THPUtils_unpackLong(PyObject* obj) {
   return (int64_t)value;
 }
 
+inline uint64_t THPUtils_unpackUInt64(PyObject* obj) {
+  unsigned long long value = PyLong_AsUnsignedLongLong(obj);
```

Comment:
Will the `unsigned long long` be big enough on all platforms?
Is there a reason not to use `uint64_t` directly here?

Reply:
Any `long long` is guaranteed to be at least 64-bit: https://stackoverflow.com/questions/5836329/how-many-bytes-is-unsigned-long-long#5836380

I could change `value` to be `uint64_t`, but that would be more of a style decision rather than a functionality decision. `PyLong_AsUnsignedLongLong()` always returns an `unsigned long long`, so either way we'd still be doing a cast--it's just a question of _when_ that cast happens. I also looked for something like `PyLong_AsUInt64()`, but it doesn't seem to exist.

Reply:
Sounds good! Thanks for the details!

Reply:
There could be merit in checking if the `unsigned long long` equals the result of casting to `uint64_t` and throw an error if they don't match due to `unsigned long long` being larger than 64 bits. That would be hard to test unless we know of and have access to a system where `unsigned long long` is larger than 64. Evidently all the CI jobs used 64-bit `unsigned long long` because they passed the test case where `torch.manual_seed(0xFFFFFFFFFFFFFFFF + 1)` is expected to throw an error.

Reply:
Yes all the CI (including rocm and windows) passed the test. So that makes me confident enough that it will work for most people. We can update that if some people with untested architectures find issues.